### ISSUE TITLE
Question/Bug: Not Addressable Entities and Path generation

### ISSUE THREAD

**Issue by sterlp** [oasis-tcs/odata-openapi#311]
Hello Guys,

we discovered an interesting use case where the XSLT generator behaves a bit odd, at least in our currently even wrong:

1. we have to entities Head and Address
2. the second entity is a dependent one, from Head, means not access-able (is-a-relation)
3. the root entity is access-able

![HeadSet ist addressierbar und beides Anlegbar - SAP](https://github.com/user-attachments/assets/5ceb88ab-d1f8-4123-a876-a575b812bb16)

As soon we add a create option for the sub entity, we also get **two** POST operation for SubEntity AddressSet. One on the parent, which is fine and one on the root. 
But it requires the link to the head and of course it's ID should not be "create able".
As a workaround we can of course set it as create able, but we end up still with two "POST" end-points, where in one the parent ID is in the body and path and in the second, only because we set it as create-able, in the body.

![HeadSet ist addressierbar und beides Anlegbar](https://github.com/user-attachments/assets/d91033c6-d717-4dbb-b23c-00638338bc97)

Overall our expectation was, that no endpoint methods are generated on top level, if we set the entity as not address able. 
Hope this description is somehow understandable.

With e.g. DELETE (or update/PATH) it works fine:
![HeadSet ist addressierbar und AddressSet löschbar - SAP](https://github.com/user-attachments/assets/2d5f2791-1260-4c0f-902d-d47172c65dd2)

OpenAPI has only the DELETE endpoint on the sub-path:
![HeadSet ist addressierbar und AddressSet löschbar](https://github.com/user-attachments/assets/c8484418-cf0e-4a61-adb1-05e235943120)

Not sure if it is a bug, or intended behavior, because of a use case I can't think of right now. I testet most combinations but this particular one somehow let us wondering.

Thanks in advance

Paul

[SAP HeadSet ist adressierbar und beides anlegbar.xml.txt](https://github.com/user-attachments/files/17294343/SAP.HeadSet.ist.addressierbar.und.beides.Anlegbar.xml.txt)

**Comment by ralfhandl** [comment]
@sterlp Hi Paul,

Thanks for reporting this!

Seems like I only considered the read case for non-addressable entity sets 🙄 

Please confirm if this is the desired behaviour:
- if AddressSet is _not_ addressable, no paths starting with `/AddressSet` will be generated
- if in addition AddressSet is creatable, a `post` operation for path `/HeadSet(...)/to_Address` will be generated
- if AddressSet is updatable / deletable, a `patch` / `delete` operation for path `/HeadSet(...)/to_Address(...)` will be generated

Thanks in advance

**Comment by sterlp** [comment]
@ralfhandl  Hello Ralf,

many thanks for the fast response. To your questions:

* if AddressSet is not addressable, no paths starting with `/AddressSet` will be generated
**=> YES**

* if in addition AddressSet is creatable, a `post` operation for path `/HeadSet(...)/to_Address` will be generated
**=> YES**

* if AddressSet is updatable / deletable, a `patch` / `delete` operation for path `/HeadSet(...)/to_Address(...)` will be generated
**=> YES** (no  `patch` / `delete` on the root `/AddressSet` )

Thanks

Paul

**Comment by ralfhandl** [comment]
@sterlp Pull request #312 fixes the first two bullet points, using a test case based on your example.

Regarding the third bullet point: I am not sure whether Gateway supports `PATCH` and `DELETE` for two-level paths with key segments such as `/HeadSet(...)/to_Address(...)`.

Could you please check with your service whether that works?

Thanks in advance!

**Comment by sterlp** [comment]
I will check on Wednesday, sorry for the delay. I think we have to add some ABAB coding to ensure the call is received or not.

**Comment by sterlp** [comment]
Hello @ralfhandl,

you are right, at least on OData v2 this won't work. Means only this works:
![image](https://github.com/user-attachments/assets/42228788-774b-4662-ab6a-90de1788f1c3)

### Good:
GET/DELETE/PATCH: `/sap/opu/odata/sap/ZE_API_DEMO_SRV/AddressSet(BusinessPartner='1',AddressId='1')`
POST: `/sap/opu/odata/sap/ZE_API_DEMO_SRV/HeadSet('1')/to_Address`

### Bad:
GET/DELETE/PATCH: `/sap/opu/odata/sap/ZE_API_DEMO_SRV/HeadSet('1')/to_Address('1')`

at least in out tests. 

----

Long story short: Only the `POST` on the `root` has to be removed as soon as the entity is not address able but create able; as otherwise we have two `POST` endpoints.

Thanks

Paul

**Comment by ralfhandl** [comment]
@sterlp Please review 
* #312

**Comment by nilskasseckert** [comment]
Hi,
I know that this issue will definitely not be fixed, but this change caused a bug. 
If you now have an OData service that only allows Create and only has a single entity, then nothing is generated anymore.
If I also make the entity addressable, I would need a Get. But my entity does not have that.

This is just some info for others who are also experiencing issues with the converter not functioning as intended. The behavior requested in the ticket only makes sense if it is a sub-entity.

**Comment by ralfhandl** [comment]
From https://sap.github.io/odata-vocabularies/docs/v2-annotations.html

> Use false if this entity set can only be accessed within its containing entity, e.g. Conditions within SalesOrders through SalesOrders(4711)/Conditions. Direct access to non-addressable entity collections will result in a 4xx response code. The set may however allow access to single entities identified by their key properties values, e.g. SalesOrderConditions(OrderID=4711,ConditionID=3)

Explicitly writing `addressable="false"` on a root entity set is an abuse of that annotation, and the outcome can be undesirable.

**Comment by nilskasseckert** [comment]
Hi @ralfhandl,
thanks for the clarification. Then option one would be the better solution if this is the standard. Thanks!
